contracts/dao/voting/vesting-lp-vault/src/contract.rs

```diff
@@ -0,0 +1,328 @@
+
+    let new_owner = deps.api.addr_validate(&new_owner)?;
+    let new_atom_vesting_lp_contract = deps.api.addr_validate(&new_atom_vesting_lp_contract)?;
+    let new_atom_oracle_contract = deps.api.addr_validate(&new_atom_oracle_contract)?;
+    let new_usdc_vesting_lp_contract = deps.api.addr_validate(&new_usdc_vesting_lp_contract)?;
+    let new_usdc_oracle_contract = deps.api.addr_validate(&new_usdc_oracle_contract)?;
+    let new_manager = new_manager
+        .map(|new_manager| deps.api.addr_validate(&new_manager))
+        .transpose()?;
+
+    if info.sender != config.owner && new_owner != config.owner {
+        return Err(ContractError::OnlyOwnerCanChangeOwner {});
+    };
+    if info.sender != config.owner
+        && new_atom_vesting_lp_contract != config.atom_vesting_lp_contract
+    {
+        return Err(ContractError::OnlyOwnerCanChangeVestingLpContract {});
+    };
+    if info.sender != config.owner
+        && new_usdc_vesting_lp_contract != config.usdc_vesting_lp_contract
+    {
+        return Err(ContractError::OnlyOwnerCanChangeVestingLpContract {});
+    };
+
+    config.owner = new_owner;
+    config.atom_vesting_lp_contract = new_atom_vesting_lp_contract;
+    config.atom_oracle_contract = new_atom_oracle_contract;
+    config.usdc_vesting_lp_contract = new_usdc_vesting_lp_contract;
+    config.usdc_oracle_contract = new_usdc_oracle_contract;
+    config.manager = new_manager;
+    config.name = new_name;
+    config.description = new_description;
+    config.validate()?;
+    CONFIG.save(deps.storage, &config)?;
+
+    Ok(Response::new()
+        .add_attribute("action", "update_config")
+        .add_attribute("description", config.description)
+        .add_attribute("owner", config.owner)
+        .add_attribute("atom_vesting_lp_contract", config.atom_vesting_lp_contract)
+        .add_attribute("atom_oracle_contract", config.atom_oracle_contract)
+        .add_attribute("usdc_vesting_lp_contract", config.usdc_vesting_lp_contract)
+        .add_attribute("usdc_oracle_contract", config.usdc_oracle_contract)
+        .add_attribute(
+            "manager",
+            config
+                .manager
+                .map(|a| a.to_string())
+                .unwrap_or_else(|| "None".to_string()),
+        ))
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn query(deps: Deps, env: Env, msg: QueryMsg) -> ContractResult<Binary> {
+    match msg {
+        QueryMsg::VotingPowerAtHeight { address, height } => Ok(to_binary(
+            &query_voting_power_at_height(deps, env, address, height)?,
+        )?),
+        QueryMsg::TotalPowerAtHeight { height } => {
+            Ok(to_binary(&query_total_power_at_height(deps, env, height)?)?)
+        }
+        QueryMsg::Info {} => query_info(deps),
+        QueryMsg::Dao {} => query_dao(deps),
+        QueryMsg::Name {} => query_name(deps),
+        QueryMsg::Description {} => query_description(deps),
+        QueryMsg::GetConfig {} => query_config(deps),
```

Comment:
mb QueryMsg::Config as every other contract?

Reply:
idk, all the vaults have very this syntax. I would leave it as is in terms of this PR. doesn't seem to be easy to be done
<img width="414" alt="image" src="https://user-images.githubusercontent.com/34917380/233312957-e96a2b92-2b37-43d8-9818-cefd4a97bbdb.png">

Reply:
Moved to #54